SDK.ipynb

```diff
@@ -78,9 +91,21 @@
    },
    "outputs": [],
    "source": [
-    "import globus_sdk\n",
+    "# fill this line in with the code that you got\n",
+    "auth_code = \"YOUR CODE HERE\"\n",
+    "\n",
+    "# and exchange it for a response object containing your token(s)\n",
+    "token_response = native_auth_client.oauth2_exchange_code_for_tokens(auth_code)\n",
```

Comment:
I ran into that error during development of the SDK features, and I think we traced it back to me creating a malformed request which probably should have been a 400.
As a result, I'm somewhat leery of that error anytime I see it -- we know that a there are things other than trying to do some unsupported grant type can trigger it. (Arguments about whether a malformed POST is "some unsupported grant type" notwithstanding.)
I'm not terribly worried that you came across it one time -- my gut call is "could not reproduce; wontfix!"


I think we should be concerned about things like double-submits, which definitely break and are very easy to do in Jupyter.
The trouble is that if they do that, the right thing to do is to start the flow over again from the start -- because their code verifier will be stale and they need to redo the `..._start_flow_...()` step.

The best I can think of is this:
```python
try:
    token_response = ...
except GlobusAPIError:
    print(("An error was encountered when exchanging your code for tokens. "
           "Please retry the entire flow, starting with the block above that "
           "generates the authorize URL"))
```

Does that seem like a good addition?

For the most part, I'm kind of inclined to optimize for success here -- I don't want to deal, for example, with the batty ways this can misfire if someone edits the `CLIENT_ID` string above.
I don't feel like I have a good grasp of the technical level of the audience for this demo, but the fact that we're presenting things in the form of notebooks at all leaves me inclined to say "not in the least bit technical", and to avoid all but the simplest error handling/recovery.